packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -17,49 +17,71 @@ import { ITestTreeProvider } from "../../utils";
 import {
 	CursorLocationType,
 	FieldKey,
+	FieldUpPath,
 	moveToDetachedField,
-	rootFieldKeySymbol,
 	UpPath,
 } from "../../../core";
 
-export type Operation = TreeEdit | Synchronize;
+export type Operation = TreeOperation | Synchronize;
+
+export type TreeOperation = TreeEdit | TransactionBoundary;
 
 export interface TreeEdit {
 	type: "edit";
-	contents: FuzzChange;
-	/** index of the tree to apply the edit to. */
-	index: number;
+	contents: FieldEdit | NodeEdit;
+}
+
+export interface Synchronize {
+	type: "synchronize";
+}
+
+export interface TransactionBoundary {
+	type: "transaction";
+	contents: FuzzTransactionEdit;
+	treeIndex: number;
+}
+
+export type FuzzFieldChange = FuzzInsert | FuzzDelete;
+
+export interface FieldEdit {
+	editType: "fieldEdit";
+	change: SequenceFieldEdit; // in the future, add `| OptionalFieldEdit | ValueFieldEdit`
 }
 
 export interface FuzzInsert {
-	fuzzType: "insert";
+	type: "insert";
 	parent: UpPath | undefined;
 	field: FieldKey;
 	index: number;
 	value: number;
+	treeIndex: number;
+}
+
+export interface SequenceFieldEdit {
+	type: "sequence";
+	edit: FuzzInsert | FuzzDelete;
 }
 
 export interface FuzzDelete extends NodeRangePath {
-	fuzzType: "delete";
+	type: "delete";
+	treeIndex: number;
+}
+
+export type FuzzNodeEditChange = FuzzSetPayload;
+
+export interface NodeEdit {
+	editType: "nodeEdit";
+	edit: FuzzNodeEditChange;
 }
 
 export interface FuzzSetPayload {
-	fuzzType: "setPayload";
+	nodeEditType: "setPayload";
 	path: UpPath;
 	value: number;
+	treeIndex: number;
 }
 
-export type FuzzChange =
-	| FuzzInsert
-	| FuzzDelete
-	| FuzzSetPayload
-	| TransactionStartOp
-	| TransactionAbortOp
-	| TransactionCommitOp;
-
-export interface Synchronize {
-	type: "synchronize";
-}
+export type FuzzTransactionEdit = TransactionStartOp | TransactionAbortOp | TransactionCommitOp;
```

Comment:
Some of these types start with "Fuzz" and others don't. It's not clear to me if the "Fuzz" prefix is meaningful. You don't have to do it now if you think it will be too many changes at once, but given that this PR is already such a huge refactor and has many renames, I would suggest removing "Fuzz" from all the types, and then either putting them in a namespace `SharedTreeFuzz` or moving them to a separate "types.ts" file and exporting them all from there.

Reply:
i agree with this suggestion with a small tweak--if you're not going to namespace them, i'd call the file `operations.ts` or `operationTypes.ts`.